pytest-embedded-nuttx/pyproject.toml

```diff
@@ -0,0 +1,41 @@
+[build-system]
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
+
+[project]
+name = "pytest-embedded-nuttx"
+authors = [
+    {name = "Filipe Cavalcanti", email = "filipe.cavalcanti@espressif.com"},
+    {name = "Fu Hanxi", email = "fuhanxi@espressif.com"},
+]
+readme = "README.md"
+license = {file = "LICENSE"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Framework :: Pytest",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python",
+    "Topic :: Software Development :: Testing",
+]
+dynamic = ["version", "description"]
+requires-python = ">=3.7"
+
+dependencies = [
+    "pytest-embedded-serial-esp~=1.11.8",
```

Comment:
I understand the `service` is optional

- `esp,nuttx` -> `NuttxEspDut`, `NuttxApp`, `NuttxSerial`
- `nuttx` -> `NuttxDut`, `Serial`

however, here it's added as a hard-dependency, You may place `pytest-embedded-serial` under `[project.dependencies`, and `pytest-embedded-serial-esp` under [project.optional-dependencies]? https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-and-requirements

Reply:
Yes, that makes more sense. Thanks!